### ISSUE TITLE
Only Gravatar image displayed in comments

### ISSUE THREAD

**Issue by davoraltman** [Automattic/jetpack#5857]
When using the Comments module, no matter which service is used to comment on a site, the Gravatar image will be displayed.

Mentioned in 2954508-t and I was able to reproduce on my test site.

**Comment by jeherve** [comment]
### Steps to reproduce

1. Log out of your Jetpack Site.
2. Leave a comment using Jetpack Comment, while logged in via Twitter.
3. Your Twitter avatar image URL is saved.
4. When displaying that comment on your site, Jetpack Comments then filters the avatar image tag that would be returned for that email address, and replaces the image URL in `src` by the saved Twitter avatar image URL:
https://github.com/Automattic/jetpack/blob/4.4.2/modules/comments/comments.php#L136-L165

While this was enough in the past, it's not enough anymore. WordPress now also adds a `srcset` value to the image tag.
https://github.com/WordPress/WordPress/blob/4.7/wp-includes/pluggable.php#L2394-L2419

We'll need to update that `srcset` value to use the saved image URL, to avoid an output like this:

```html
<img alt="" src="https://i2.wp.com/abs.twimg.com/sticky/default_profile_images/default_profile_0_normal.png?resize=68%2C68" srcset="http://2.gravatar.com/avatar/22544f7feb1a482f933e9c768b99c3db?s=136&amp;d=mm&amp;r=g 2x" class="avatar avatar-68 photo" scale="0" width="68" height="68">
```

5. If we don't update this, users on a HIDPi device see the image pulled from the `srcset` value.

**Comment by tycooko** [comment]
Hello. Is someone still working on this issue? The problem has not been solved.

**Comment by jeherve** [comment]
@tycooko We haven't made any progress on this yet, but we'll update this issue when we do!

**Comment by kraftbj** [comment]
We should switch to a different filter (added in WP 4.2): https://developer.wordpress.org/reference/hooks/get_avatar_data/

We can just filter the URL instead of trying to fix up the `img src`.

**Comment by lamdayap** [comment]
reporte here 3250576-t 

I did some digging around and noticed that the issue is only affecting Twitter at this stage. Facebook appears to be pulling the correct profile image, however Twitter is pulling a broken link in **src**:

`https://i0.wp.com/pbs.twimg.com/profile_images/803578876450344961/3QTEAyiv_normal.jpg?resize=34%2C34`

Similar to what Jeremy found.

At the same time this is being passed in **srcset**:

`https://secure.gravatar.com/avatar/98f14ff7e4725b28acfa720085c3a4ff?s=68&d=mm&r=g`

Which is another broken link

**Comment by tycooko** [comment]
Hi lamdayap. It is not pulling the right Facebook image for me, as you can see in the comments section of this post: [http://www.psmtickling.com/blog/tickle-test-on-spanish-tv/#comments](http://www.psmtickling.com/blog/tickle-test-on-spanish-tv/#comments) .
The first profile pic doesn't appear (whereas the Facebook user does have a public profile pic); the second one, I don't even know from where it pulls it (the Facebook user has a different pic).
There are issues with Google+, too: the third comment was posted with a Google+ account, but, since the email address was the same as the one of the first comment, Jetpack shows the same (wrong) profile pic and links to the same Facebook profile.
It's like Jetpack doesn't really see and respect which profile the user wants to use; once it registers in its "cache" an email address, it connects that to a profile and one only, and forces the user to use that profile.

**Comment by seejacobscott** [comment]
3260467-t 
From user: 

> When a user leaves a comment using facebook the avatar for that user does not
> load.
> 
> Upon inspecting the source of the comment I see this:
> 
> <Code Stripped out by email system :( >
> 
> Changing the first line to https://graph.facebook.com fixes the issue for me but
> that's only with the web browser debugger.
> 
> Is there a solution for this? Thanks.

**Comment by seejacobscott** [comment]
Further update from the above user: 

> cloudflare users can enable "Automatic HTTPS Rewrites"
> and it seems to fix the issue as it's forcing everything to re-write to
> https.

**Comment by aheckler** [comment]
3136104-t

Bumping priority to Normal since this is affecting more people.

**Comment by lamdayap** [comment]
[https://wordpress.org/support/topic/jetpack-comments-dont-work/](https://wordpress.org/support/topic/jetpack-comments-dont-work/)

Facebook not pulling profile iamge

**Comment by seejacobscott** [comment]
Seem to be seeing the same thing with a Jetpack Business Concierge user, follow up ticket is here:
781418-zen

Tried commenting on the site with a Facebook profile, but none showed. Older posts and newer posts seem to have different reasons for failing though. 

Did a little digging, on their older posts the images are still trying to pull from Facebook, but seem to have a bad link, formatting like this: 

`<img alt="" src="http://graph.facebook.com/jetpack.jacob/picture" class="avatar avatar-50 photo" height="50" width="50">`

the response is this: 

```
{
   "error": {
      "message": "(#803) Cannot query users by their username (jetpack.jacob)",
      "type": "OAuthException",
      "code": 803,
      "fbtrace_id": "CAvh8n2kNZc"
   }
}

```

With more recent posts, it's trying to pull a gravatar, and doesn't always seem to work unless it's authorizing for wp.com for the comment.

**Comment by hirushacooray** [comment]
I'm also having this issue and found this thread. Is there a fix for this issue?

**Comment by hirushacooray** [comment]
I just inspected the src of image. If I remove the i0.wp.com part from the URL, then profile picture is loading fine.

**Comment by jeherve** [comment]
@hirushacooray We have not made any progress on this yet, but we'll update this issue when we do. Until then, I would recommend that you either deactivate avatars under Settings > Discussion in your dashboard, or by deactivating Jetpack's Comments feature under Jetpack > Settings > Traffic.

**Comment by hirushacooray** [comment]
@jeherve Thanks. I wondering why jetpack is still try to grab images via CDN, since that module is disabled. (/wp-admin/admin.php?page=jetpack_modules).

**Comment by jeherve** [comment]
Jetpack's Comment feature relies on the CDN to display its avatars, whether the CDN feature is on or not. This allows us to display Retina / HiDPi friendly avatars for everyone. That's certainly something we could change in the future.

**Comment by hirushacooray** [comment]
Is there any temporary fix for this? I'm trying to fix it for hours but no luck :worried:

**Comment by jeherve** [comment]
You could deactivate avatars under Settings > Discussion in your dashboard, or deactivate Jetpack's Comments feature under Jetpack > Settings > Traffic.

**Comment by hirushacooray** [comment]
yes. That's right. But I mean is there any temporary fix to solve this and display fb profile pic?

**Comment by jeherve** [comment]
I am afraid not. We'll update this issue once we find a way.

**Comment by hirushacooray** [comment]
No worries. Hope this will be fixed in the future.

**Comment by mikeicode** [comment]
Ran into a similar issue 3722027-zd-woothemes the site was AT reverted and then the Twitter and Facebook gravatar images didn't work.

The missing images link to `i0.wp.com/pbs.twimg.com` and `i2.wp.com/graph.facebook.com`. When they view the same comments on a copy of the site on a local install the images work fine, they link to `2.gravatar.com`.
